custom_components/elastic.py

```diff
@@ -238,11 +250,13 @@ def enqueue_state(self, entry):
         domain = state.domain
         entity_id = state.entity_id
 
-        if domain in self._excluded_domains:
+        if (not self._included_domains and domain in self._excluded_domains) 
+            or (self._included_domains and domain not in self._included_domains):
             _LOGGER.debug("Skipping %s: it belongs to an excluded domain", entity_id)
             return
 
-        if entity_id in self._excluded_entities:
+        if (not self._included_entities and entity_id in self._excluded_entities) 
+            or (self._included_entities and entity_id not in self._included_entities):
             _LOGGER.debug("Skipping %s: this entity is explicitly excluded", entity_id)

```

Comment:
These log messages aren't technically correct in all cases. We'll want to either rephrase these messages, or have dedicated messages for the include and exclude scenarios.

Reply:
I took out the function and call it in `enqueue_state` and `do_publish`.